frontend/awx/administration/instances/hooks/useSelectAssociatePeers.tsx

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { MultiSelectDialog, usePageDialog } from '../../../../../framework';
+import { usePeersTabFilters } from '../Instances';
+import { awxAPI } from '../../../common/api/awx-utils';
+import { useAwxView } from '../../../common/useAwxView';
+import { useAssociatePeersToInstance } from './useAssociatePeersToInstance';
+import { usePeersToAssociateColumns } from './usePeersColumns';
+import { useGetItem } from '../../../../common/crud/useGet';
+import { useParams } from 'react-router-dom';
+import { Peer } from '../../../interfaces/Instance';
+
+function SelectAssociates(props: {
+  accessUrl?: string;
+  title: string;
+  onSelect: (peers: Peer[]) => void;
+  confirmText?: string;
+}) {
+  const { t } = useTranslation();
+  const params = useParams<{ id?: string }>();
+  const { data: instance } = useGetItem<Peer>(awxAPI`/instances/`, params.id);
+
+  const toolbarFilters = usePeersTabFilters();
+  const columns = usePeersToAssociateColumns(undefined);
+  const tableColumns = useMemo(
+    () =>
+      columns.filter((item) =>
+        ['Instance name', 'Address', 'Port', 'Node type', 'Protocol'].includes(item.header)
+      ),
+    [columns]
+  );
+
+  const instanceName = instance?.hostname as string;
+  const view = useAwxView<Peer>({
```

Comment:
Should this be ReceptorAddress instead of Peer?

Reply:
Peer is correct - actually i removed `ReceptorAddress` type